File: Category/Two Pointers/0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size() - 1;
        int ret = 0;
        while (left < right)
        {
            ret = max(ret, min(height[left], height[right]) * (right - left));
            if (height[left] <= height[right])
                left += 1;
            else
                right -= 1;
        }
        return ret;
    }
};

File: Problems/0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r = 0, len(height) - 1
        max_volume = 0
        while l < r:
            if height[l] < height[r]:
                max_volume = max(max_volume, height[l] * (r-l))
                l+=1
            elif height[l] >= height[r]:
                max_volume = max(max_volume, height[r] * (r-l))
                r-=1
        return max_volume